trigger/resource/resource.go

```diff
@@ -77,7 +78,12 @@ func NewTrigger(logger log.Logger, client *client.Client, configFile string) (*T
 	}
 
 	for _, r := range config.Resources {
-		c, err := client.ClientFor(r.APIVersion, r.Kind, "")
+		key := r.Name
+		if len(r.Namespace) > 0 {
+			key = r.Namespace + "/" + r.Name
```

Comment:
Actually on second thought, I don't think this is needed. The user is responsible for making sure these don't clash, `r.Name` is just any name that the user wants to give this resource-source.

Do you mind reverting this part?

We should just check that `t.infs[key]` doesn't already exist when we write it to that map.

Reply:
ahh, yes this makes sense, so the name is completely user-defined. yes, let me revert it in a follow-up.